feat: added dart sass support

PeculiarGoat opened the pull request:
This PR depends on the [core-template one](https://github.com/shift72/core-template/pull/166) being merged first.

- Bumped kibble version.
- Moved from Node Sass to Dart Sass.
- Copy scss files to output instead of concatenate them to main scss.  This means we can make full use of Sass features like having multiple files with imports etc.

Issues:
- Can't figure out a way to stop Sass printing warnings for Bootstrap without disabling all warnings.
- The npm scripts for this repo are pretty messy now.  When you run the site, Sass runs multiple times which results in the same warnings getting printed multiple times... not nice to look at.  Might be a good idea to look at reworking those npm scripts.... there's a good chance that I haven't covered all cases with the web of watching and building.

stajs commented on package.json:
```diff
@@ -22,7 +22,7 @@
     "css:local:backup": "cp ./output/site/styles/main.scss ./output/site/styles/_main.scss",
     "css:local:variables": "concat -o ./output/site/styles/_variables.scss ./local/site/styles/_variables.scss ./node_modules/@shift72/core-template/site/styles/_variables.scss && npm run css",
-    "css:local:custom": "concat -o ./output/site/styles/main.scss ./output/site/styles/_main.scss ./local/site/styles/local.scss && npm run css:local:variables",
+    "css:local:custom": "cp -RT ./local/site/styles ./output/site/styles/local && concat -o ./output/site/styles/main.scss ./output/site/styles/_main.scss && npm run css:local:variables",
```
I'm having trouble understanding the `concat` now. Previously it was:

```
concat
   -o ./output/site/styles/main.scss  (output file)
      ./output/site/styles/_main.scss (input 1)
      ./local/site/styles/local.scss  (input 2)
```

Now it is (?):

```
concat
  -o ./output/site/styles/main.scss  (output file)
     ./output/site/styles/_main.scss (just the one input?)
```

If that is intended:

1. Why `concat` and not just rename (or `cp`)?
2. Do we not have a `local.scss` anymore? How are things overriding?

stajs replied:
And I know you said "Copy scss files to output instead of concatenate them to main scss" but I'm missing something.

PeculiarGoat replied:
We couldn't `@import` or `@use` in `local.scss` is because it was concatenated to `main.scss` before.

Now `local.scss` (and any other files in that directory) are copied instead to `./output/site/styles/local` and the `main.scss` already has an `@import 'local/local'` statement.  My thinking behind this is:
1. We can use `@import` and `@use` in `local.scss`.
2. We can name our local scss files whatever we want without worrying about clashing with the ones in core template because they are tucked into that `local` directory.

We just have to enforce a rule that `local/local` is the last import in `main.scss` to guarantee that it overrides (in the same way that concatenating to the end of the file does).

stajs replied:
I understand the change to allow `local.scss` to `@import`/`@use`, but I'm not sure it's answered my first question:

> Why concat and not just rename (or cp)?

For example, instead of `concat -o main.scss _main.scss` just `cp -f _main.scss main.scss`? And it begs the question if we aren't concatentaing multiple files anymore, do we even need `_main.scss` - why not just have `main.scss`?

The reason why I'm digging in on `concat` is to see if we can potentially remove a dependency.

PeculiarGoat replied:
Ah gotcha... yeah I think we can probably ditch `concat`.  I'll keep that in mind while I sort out a js tool to run sass smarter.

stajs reviewed (commented):
It's late and I need some explaining...

stajs commented:
How much effort to not run sass multiple times?

PeculiarGoat commented:
> How much effort to not run sass multiple times?

For the record, node-sass already runs three times every time you start the site... it just doesn't print out the warnings so it's less offensive.

I'm thinking perhaps I could add some css handling script(s) to `tools` so it can be done a bit smarter and make `package.json` easier to read.

Thoughts @stajs ?

stajs commented:
@PeculiarGoat sounds good, and happy to deal with that improvement separately/later.

PeculiarGoat commented:
@stajs Think this is done now.

- Swapped `node-sass` for `sass`.
- Bumped `core-template` version.
- Bumped `kibble` version.
- Removed `concat` dependency.
- Added `css.js` script to handle local `scss` copying/concatenating and running of `sass`.
- Big refactor of `package.json` css scripts.  Much simpler and human readable now.

Potential future extra work:
- Refactor the js scripts in `package.json` because they don't work as well as they should.

stajs commented on package.json:
```diff
@@ -4,28 +4,24 @@
+    "start": "npm run init && npm-run-all --parallel js:watch js:watch:local css:watch kibble:watch kibble:watch:local json:kibble:lang:watch",
+    "init": "node tools/build-output.js --mode=init && npm-run-all css js merge:json:kibble merge:json:lang",
     "build": "npm run kibble:build",
-    "publish": "npm run build",
```
Can we keep this alias? This is the one I use and what I recommend to clients to use. The reason being in my mind "publish" is more akin to deploying the site, while "build" doesn't carry that connotation - for me I would expect a "build" to be a local thing only.

It also ties to `npm publish` (to publish a package).

In fact, I would be happy if `kibble:build` didn't have the `&& kibble publish` and we separated them out, then `publish` could be `npm run kibble:build && kibble publish`.

stajs reviewed (changes requested):
Just one thing!
